### PR TITLE
Add color schemes to color selection options

### PR DESCRIPTION
**Pull request by ppillot**
Color selection scheme can accept color schemes (eg `sstruc`)
`scheme = NGL.ColormakerRegistry.addSelectionScheme([["red", "1-10"],["sstruc","11-30"]],"myScheme")`
a third parameter can be added in the pair list
`scheme = NGL.ColormakerRegistry.addSelectionScheme([["red", "1-10"],["uniform", "11-30", {value: 16777215}]], "myScheme")`

### CODE REVIEW

File: src/color/selection-colormaker.js [hunk 1]
```diff
@@ -21,18 +22,37 @@ class SelectionColormaker extends Colormaker{
...
+                    structure: this.structure, 
+                    value: 0x909090 
+                }
+                :
+                { 
+                    scheme: 'uniform',
+                    value: new Color( pair[ 0 ] ).getHex() 
+                };
+            
+            let param = pair[2] || {};
```
Comment (arose) [inline src/color/selection-colormaker.js]: not need, Object.assign handles `undefined`

File: src/color/selection-colormaker.js [hunk 2]
```diff
@@ -21,18 +22,37 @@ class SelectionColormaker extends Colormaker{
...
+            Object.assign( schemeProperties, param )
+
+            this.colorList.push( ColormakerRegistry.getScheme( schemeProperties ) );
```
Comment (arose) [inline src/color/selection-colormaker.js]: please rename to colormakerList

File: src/color/selection-colormaker.js [hunk 3]
```diff
@@ -21,18 +22,37 @@ class SelectionColormaker extends Colormaker{
 
         this.colorList = [];
         this.selectionList = [];
+        
+        this.schemesList = ColormakerRegistry.getSchemes();
```
Comment (arose) [inline src/color/selection-colormaker.js]: please add a `hasScheme` to the colormaker-registry and use that

```
hasScheme( id ){
    return id in this.schemes || id in this.userSchemes; 
}
```

**Comment by arose** [comment]
Could you please also change the 'selectionColoring' example in `examples/js/examples.js` to show how to use the new feature.

**Comment by ppillot** [comment]
All done !

**Comment by arose** [comment]
thank you @ppillot 

just slightly tweaked it trying out some array destructuring (never used it before), see 9fcc39ce15f1cf23ea1660812d1a7b5479fa4686

**Pull request merged**